moto/emr/utils.py

```diff
@@ -37,3 +38,109 @@ def steps_from_query_string(querystring_dict):
             idx += 1
         steps.append(step)
     return steps
+
+
+class Unflattener:
+    @staticmethod
+    def unflatten_complex_params(input_dict, param_name):
+        """ Function to unflatten (portions of) dicts with complex keys.  The moto request parser flattens the incoming
+        request bodies, which is generally helpful, but for nested dicts/lists can result in a hard-to-manage
+        parameter exposion.  This function allows one to selectively unflatten a set of dict keys, replacing them
+        with a deep dist/list structure named identically to the root component in the complex name.
+
+        Complex keys are composed of multiple components
+        separated by periods. Components may be prefixed with _, which is stripped.  Lists indexes are represented
+        with two components, 'member' and the index number. """
```

Comment:
Proper documentation is always good to see :+1:

Reply:
Tx!